### ISSUE TITLE
Does not work on Google Chrome in Ubuntu

### ISSUE THREAD

**Issue by Lorempn2** [PhilGrayson/chrome-csp-disable#7]
I activated the plugin but i keep getting policiy errors (Refused to connect to..). The policy is set via the response header, I want to disable the policiy.

**Comment by PhilGrayson** [comment]
What URL are you browsing to? And can you attach a screenshot or copy/paste the errors you are seeing?

**Comment by Lorempn2** [comment]
I am playing with web.whatsapp.com<http://web.whatsapp.com>, a post or get request is not working (tried it by executing an ajax request via the dev-console in chrome)

Error:

Refused to connect to 'http://test.com' because it violates the following Content Security Policy directive: "connect-src 'self' wss://*.web.whatsapp.com https://*.whatsapp.net https://dyn.web.whatsapp.com https://*.giphy.com https://*.tenor.co blob: https://crashlogs.whatsapp.net/wa_clb_data https://crashlogs.whatsapp.net/wa_fls_upload_check".

**Comment by PhilGrayson** [comment]
I'm able to replicate the issue.
1. Browse to https://web.whatsapp.com/
2. Run in Javascript Console
```
var xhr = new XMLHttpRequest();
xhr.open('GET', 'http://example.com', true);
```
3. See error;
```
VM2088:1 Refused to connect to 'http://example.com/' because it violates the following Content Security Policy directive: "connect-src 'self' wss://*.web.whatsapp.com https://*.whatsapp.net https://dyn.web.whatsapp.com https://*.giphy.com https://*.tenor.co blob: https://crashlogs.whatsapp.net/wa_clb_data https://crashlogs.whatsapp.net/wa_fls_upload_check".
```

**Comment by PhilGrayson** [comment]
Either this plugin's initial permissions ([manifest.json](https://github.com/PhilGrayson/chrome-csp-disable/blob/master/manifest.json#L12-L13)) OR the permissions to attach an `onHeadersReceived` ([background.js](https://github.com/PhilGrayson/chrome-csp-disable/blob/master/background.js#L28)) listener are wrong. 

Initial look at them seem okay.

**Comment by PhilGrayson** [comment]
Very much related #2

**Comment by Lorempn2** [comment]
Yep, indeed. It all looks correct.. I'm not understanding why this issue occurres. I also tried creating my own plugin to see what headers i'm receiving via `onHeadersReceived` in a background.js script. I received all headers, except the content-policy header. Also noticed that the `url` property was never equal to the root domain it self.

**Comment by Lorempn2** [comment]
Found out that only GET requests are not working, POST is working here.

**Comment by PhilGrayson** [comment]
I'm not sure why https://web.whatsapp.com is being treated differently for this extension. https://www.whatsapp.com is behaving as expected.

I find it odd that POSTs are working for you. How are you testing that? `xhr.open('POST', 'http://example.com', true);` gives me the same errors on web.whatsapp.com.

**Comment by Lorempn2** [comment]
If i do a post via the console, I indeed get the same error. If i execute the post via an chrome extension, it works. Something to do with permissions. The strange thing is that a get request (via a chrome extension is NOT working..)

**Comment by Lorempn2** [comment]
Maybe it is because web.whatsapp.com is using lowercased headers (content-security-policy) and www.whatsapp.com is using camelcased headers (Content-Security-Policy).

Edit: should not make a difference since the plugin is casting names to lowercase.

`if ('content-security-policy' === details.responseHeaders[i].name.toLowerCase()) {`

**Comment by PhilGrayson** [comment]
When putting a breakpoint [on this line](https://github.com/PhilGrayson/chrome-csp-disable/blob/master/background.js#L4), that never gets triggered when on https://web.whatsapp.com. It does get triggered when browsing to any other page I tested.

I'm currently at $DAY_JOB. I'll take another look when I get home.

**Comment by PhilGrayson** [comment]
The document returned from https://web.whatsapp.com/ is not counted as either a main_frame or sub_frame, failing [this filter](https://github.com/PhilGrayson/chrome-csp-disable/blob/master/background.js#L29).

Non of the supported [ResourceTypes](https://developer.chrome.com/extensions/webRequest#type-ResourceType) match that document.

These requests I'm able to capture; 
```
type: url
-------------
other: https://web.whatsapp.com/404.appcache
other: https://web.whatsapp.com/serviceworker.js
stylesheet: https://fonts.googleapis.com/css?family=Open+Sans:400,600%7CRoboto:300,400,500
font: https://fonts.gstatic.com/s/roboto/v16/oMMgfZMQthOryQo9n22dcuvvDin1pK8aKteLpeZ5c0A.woff2
font: https://fonts.gstatic.com/s/roboto/v16/RxZJdnzeo3R5zSexge8UUZBw1xU1rKptJj_0jans920.woff2
font: https://fonts.gstatic.com/s/roboto/v16/Hgo13k-tfSpn0qi1SFdUfZBw1xU1rKptJj_0jans920.woff2
websocket: wss://w6.web.whatsapp.com/ws
font: https://fonts.gstatic.com/s/opensans/v13/cJZKeOuBrn4kERxqtaUH3ZBw1xU1rKptJj_0jans920.woff2
font: https://fonts.gstatic.com/s/opensans/v13/MTP_ySUJH_bn48VBG8sNShampu5_7CjHW5spxoeN3Vs.woff2
image: https://web.whatsapp.com/img/favicon/1x/favicon.png
```

**Comment by PhilGrayson** [comment]
The document has (from ServiceWorker) when looking at it in the Network tab, indicating it isn't being loaded 'normally'.

<img width="540" alt="screen shot 2017-05-24 at 20 29 33" src="https://cloud.githubusercontent.com/assets/143456/26421969/f3b667a0-40bf-11e7-86fb-fbec674c19cf.png">

**Comment by PhilGrayson** [comment]
It's the [Application Cache](https://developer.mozilla.org/en-US/docs/Web/HTML/Using_the_application_cache) that is getting in the way.

To verify;
1. Disable Content Security Policy using this extension
2. Browse to http://web.whatsapp.com
3. Run
```
var jq = document.createElement('script');
jq.src = "https://code.jquery.com/jquery-3.2.1.min.js";
document.getElementsByTagName('head')[0].appendChild(jq)
```
See this error
```
VM8167:3 Refused to load the script 'https://code.jquery.com/jquery-3.2.1.min.js' because it violates the following Content Security Policy directive: "script-src 'self' 'unsafe-eval' https://ajax.googleapis.com".
```
4. Clear site data for web.whatsapp.com. Settings -> Content Settings... (behind '+ Show advanced settings') -> All cookies and site data... -> search for whatsapp

<img width="740" alt="screen shot 2017-05-24 at 20 57 13" src="https://cloud.githubusercontent.com/assets/143456/26422948/9f15aa04-40c3-11e7-9161-2209d081c9ba.png">

5. Browse to https://web.whatsapp.com/ again
6. Run 
```
var jq = document.createElement('script');
jq.src = "https://code.jquery.com/jquery-3.2.1.min.js";
document.getElementsByTagName('head')[0].appendChild(jq)
```
See the Content Security Policy error does not get triggered.

**Comment by PhilGrayson** [comment]
I've pushed a partial fix for this issue to v1.0.6 of this extension.

When you disable CSP by clicking the extension button, the Application Cache is cleared. The next time you refresh http://web.whatsapp.com, CSP will be disabled.

Unfortunately if you refresh the page again, CSP will become enabled. This is because the cache has been rebuild. I could work around it with a timer that periodically clears the cache if CSP is disabled.
